### PR TITLE
Minor update to Struts 2.5.x build configuration, SecurityMemberAccess.

### PR DESCRIPTION
**Pull request by JCgH4164838Gh792C124B5**
Minor update to Struts 2.5.x build configuration, SecurityMemberAccess.

pom.xml changes:
- updated `maven-surefire-plugin` 2.20.1-> 2.22.1

code changes:
- updated `SecurityMemberAccess` for compatibility with future OGNL 3.1.19+ versions
- updated `SecurityMemberAccess` constructor (minor cleanup and comment)

### CODE REVIEW

**Comment by JCgH4164838Gh792C124B5** [comment]
Hello.
Although this PR should permit usage of OGNL 3.1.19+, _it may be prudent to stick with 3.1.18 for the Struts 2.5.19 build_.  There are enough changes between 3.1.15-to-3.1.18 to absorb, and the behaviour changes in 3.1.19 may need some deeper consideration before including in a future Struts build.

**Comment by yasserzamani** [comment]
I personally think those null checks and throwing exceptions may aren't essential because they decrease our test coverage percent and also they're for Struts internal usages so almost always aren't null. I also had an improvement at 459b3f7023176a516b8cec26daa47c4b1902ebed which you may like to see and import :) Thanks!

**Comment by JCgH4164838Gh792C124B5** [comment]
Hello @yasserzamani .

I've taken a quick look at the changes you made with https://github.com/apache/struts/commit/459b3f7023176a516b8cec26daa47c4b1902ebed for `WW-4973` ( resolved for `Struts 2.6`).  They look good but are more sophisticated/involved than I was thinking of for this PR.   :)

So I attempt to remove the null checks/explicit exception throws you pointed out and try pushing that up for further review (should resolve the code coverage issue).  

Let me know what you think of the revised changes (hopefully "good enough" for future OGNL 3.1.x in 2.5.x, without too many other changes).

File: pom.xml [hunk 1]
```diff
@@ -236,12 +236,12 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>${maven-surefire-plugin.version}</version>
```
Comment (sepe81) [inline pom.xml]: > Since the conflict didn't show anything related to the new variable for surefire I think I made the correct choice ...

The previous change extracted `<maven-surefire-plugin.version>2.22.1</maven-surefire-plugin.version>` for better reuse. IMO you should better revert this change for `pom.xml` b/c the dependency is already up-to-date.

Comment (sepe81) [reply pom.xml]: @JCgH4164838Gh792C124B5 any help needed for the rebase/amend?

Comment (JCgH4164838Gh792C124B5) [reply pom.xml]: Hello @sepe81 .
I'll try to rebase/amend the change to pom.xml, but it's not something I've done before with Git.  If I don't post an update in the next day or two it's a pretty safe bet I'm stumped.  :)

**Comment by JCgH4164838Gh792C124B5** [comment]
After attempting to amend/rebase my change locally I encountered some Git behaviour such that I'm not trusting pushing back to my remote fork.  
So ... I will instead close this PR, re-do my (now single file) change in a new branch and create a new PR.

**Comment by sepe81** [comment]
I use it all the time with an alias as `git please`. 

see https://medium.freecodecamp.org/git-please-a182f28efeb5

**Comment by JCgH4164838Gh792C124B5** [comment]
Thank you very much for the advice, @lukaszlenart  and @sepe81 .  Next time I'll be less worried about the state of my remote working branch.  :)
I have added the git alias (thanks to the instructions from the link above), and hopefully I'll remember I have it now.  ;-)
